### ISSUE TITLE
Feature Request: Disable all notifications when playing a movie

### ISSUE THREAD

**Issue by S-UP** [lhc70000/iina#1368]
**System and IINA version:**

- macOS 10.13.2
- IINA 0.0.15

**Expected behavior:**

I would suggest you disable all system notifications and application updates when playing a movie. I don't want to see a system notification about an incoming email or an update to the video player. The goal should be that while playing a movie the user of the player is not distracted at all. One should be able to concentrate fully on the movie.

**Comment by saagarjha** [comment]
If this ends up being a commonly requested feature, I think we can reconsider, but as it is now I think this is too niche of a feature to be implemented. The API for Do Not Disturb is private (though, it's not too difficult to work with), and I can't think of any other apps that do this. For your case I'd think it would be better if this was done separately from IINA (i.e. write a script of some sort that watches for IINA to launch, then disables notifications).

**Comment by whoinoi** [comment]
I strongly object again implementing this. There are a lot of use cases of IINA. Not everybody watches a movie and not everybody wants to *not* be disturbed.

Those that don't want to be disturbed should turn on silent mode themselves. It's easily accessible in the notification center with just a few clicks.

There are a lot of important things to work on for the developers of IINA. This is not one of it.

**Comment by damengilland** [comment]
Option + Click on the notification icon in the menu bar will temporarily disable notifications.

BUT... I really would like to have the option to automatically disable notifications when movies are playing.

Sadly, so many things (looking at you, Outlook) don't use the Mac OS notifications and would still pop up.

**Comment by whoinoi** [comment]
@damengilland Ever thought about turning of your Internet connection during a movie? It's even less clicks than to reach the Do Not Disturb setting.

Also there are dedicated apps out there which allow you to select when you want to get notifications. IINA shouldn't intervene here, because there are a lot more use cases to IINA than just undisturbed movie watching.

**Comment by megadr01d** [comment]
To all the yayers and the nayers, Movist disables notifications **only** when playing fullscreen and I think it makes sense.

**Comment by saagarjha** [comment]
As I've mentioned above, I generally consider handling Do Not Disturb and notifications outside the scope of IINA. Unless there's a significant demand for it, we will probably prioritize other features over this one. For those looking for this functionality, I'd suggest writing an AppleScript or similar that replicates this functionality.

**Comment by Stanzilla** [comment]
@inflation Can we reopen this then please?

**Comment by Stanzilla** [comment]
@saagarjha can we please reopen this? It should be an option since not everyone will want it.

**Comment by comomac** [comment]
To all who want to watch movie peacefully without getting disturbed by the annoying popup notification, I've figured out another way to do it.

Launch System Settings then Focus section, I've create a focus profile called "Videos" then inside that profile, I've set "none" for Allow Notifications, then add schedule when using ("IINA") App.

As long as IINA is in focus, the MacOS Focus aka DND (Do Not Disturb) will be activated. If you quit IINA or focus other window, it will disable DND again.

Hope this helps, enjoy.

![Image](https://github.com/user-attachments/assets/592eff17-1eef-4a9e-8090-c0d0898f6a82)
